### PR TITLE
Added factories to produce social auth for Users

### LINKED ISSUE #3166: UserFactory does not create social auth
This means some data (e.g. ProfileSerializer) generates with `None` values.

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?
Closes #3166 

#### What's this PR do?
Adds factories to produce social auth for Users, and changes a lot of test cases to use those factories

#### How should this be manually tested?
No manual testing needed

#### Where should the reviewer start?
`micromasters/factories.py` and `profiles/factories.py`

#### Any background context you want to provide?
- Here's some example usage for the new factories:
    ```python
    # Create a user with a social auth provider 'abc'
    SocialUserFactory.create(username='user_0', social_auth__provider='abc')
    # Same as above, but create a Profile instead of a User
    SocialProfileFactory.create(first_name='Joe', user__username='user_0', user__social_auth__provider='abc')
    ```
- There are a bunch of other test cases that manually add social_auth records on their own, but this PR has enough lines as it is

### CODE REVIEW

File: micromasters/factories.py [hunk 1]
```diff
@@ -14,3 +45,27 @@ class UserFactory(DjangoModelFactory):
...
     class Meta:
         model = User
+
+
+class UserSocialAuthFactory(DjangoModelFactory):
+    """Factory for UserSocialAuth"""
+    user = SubFactory(UserFactory)
+    provider = EdxOrgOAuth2.name
+    extra_data = {"access_token": "fooooootoken"}
+    uid = LazyAttribute(lambda social_auth: social_auth.user.username)
```
Comment (noisecapella) [inline micromasters/factories.py]: Can you change this to be different from `user.username`? The `_edx` suffix was originally added to help make it obvious if we leaked the Django username through the API instead of using the edX one

File: micromasters/factories.py [hunk 2]
```diff
@@ -2,9 +2,40 @@
...
+
+    Returns:
+        dict: A dict of kwargs related to a specific model (without the model prefix)
+    """
+    related_model_prefix = '{}__'.format(related_model_prop_name)
+    extracted_kwargs = pop_matching_keys_from_dict(
+        orig_kwargs,
+        lambda key: key.startswith(related_model_prefix)
+    )
+    return {k.replace(related_model_prefix, ''): v for k, v in extracted_kwargs.items()}
```
Comment (noisecapella) [inline micromasters/factories.py]: This might do something unexpected if `related_model_prefix` exists twice in the prop name. Can you change this to `replace(related_model_prefix, '', 1)` and add a test for this edge case?

File: profiles/permissions_test.py [hunk 3]
```diff
@@ -73,23 +74,12 @@ class CanSeeIfNotPrivateTests(MockedESTestCase):
     def setUp(self):
         super(CanSeeIfNotPrivateTests, self).setUp()
         with mute_signals(post_save):
```
Comment (noisecapella) [inline profiles/permissions_test.py]: Do we still need `mute_signals` here and below?

File: micromasters/factories.py (hunk 2 above)
Comment (gsidebo) [inline micromasters/factories.py]: good catch 👍
